File: UI/src/app/modules/admin/pages/movie-processing/models/requests/movie-processing-week-request.model.ts
import {MovieProcessingDayRequestModel} from './movie-processing-day-request.model';

export class MovieProcessingWeekRequestModel {
  public weekNumber: number;
  public days: MovieProcessingDayRequestModel[];

  constructor() {
    this.days = [];
    for (let dayIndex = 0; dayIndex < 7; dayIndex++) {
      this.days.push(new MovieProcessingDayRequestModel());
    }
  }
}

File: API/src/main/java/com/MateuszLebioda/OneCinema/Model/ScreeningRoom/MovieProcessingWeekRequestModel.java
package com.MateuszLebioda.OneCinema.Model.ScreeningRoom;

import java.util.Set;

public class MovieProcessingWeekRequestModel {
   private int weekNumber;
   private Set<MovieProcessingDayRequestModel> days;

    public boolean validate(){
        return  validateWeekNumber() &&
                validateEveryOneDay();

    }

    private boolean validateEveryOneDay(){
        for(MovieProcessingDayRequestModel mode:days){
            if(!mode.validate()){
                return false;
            }
        }
        return true;
    }

    private boolean validateDays(){
        return days.size() > 0;
    }

    private boolean validateWeekNumber(){
        return  weekNumber >= 0;
    }

    public int getWeekNumber() {
        return weekNumber;
    }

    public void setWeekNumber(int weekNumber) {
        this.weekNumber = weekNumber;
    }

    public Set<MovieProcessingDayRequestModel> getDays() {
        return days;
    }

    public void setDays(Set<MovieProcessingDayRequestModel> days) {
        this.days = days;
    }
}
